joint_trajectory_controller/test/test_trajectory_controller.cpp

```diff
@@ -441,3 +441,82 @@ TEST_F(TestTrajectoryController, correct_initialization_using_parameters) {
   ASSERT_EQ(State::PRIMARY_STATE_INACTIVE, state.id());
   executor.cancel();
 }
+
+void test_state_publish_rate_target(
+  std::shared_ptr<joint_trajectory_controller::JointTrajectoryController> traj_controller,
+  int target_msg_count)
+{
+  // fill in some data so we wont fail
+  auto traj_lifecycle_node = traj_controller->get_lifecycle_node();
+  std::vector<std::string> joint_names = {"joint1", "joint2", "joint3"};
+  rclcpp::Parameter joint_parameters("joints", joint_names);
+  traj_lifecycle_node->set_parameter(joint_parameters);
+
+  std::vector<std::string> operation_mode_names = {"write1", "write2"};
+  rclcpp::Parameter operation_mode_parameters("write_op_modes", operation_mode_names);
+  traj_lifecycle_node->set_parameter(operation_mode_parameters);
+
+  rclcpp::executors::SingleThreadedExecutor executor;
+  executor.add_node(traj_lifecycle_node->get_node_base_interface());
+
+  traj_lifecycle_node->set_parameter(
+    rclcpp::Parameter(
+      "state_publish_rate",
+      static_cast<double>(target_msg_count)));
+
+  traj_controller->on_configure(traj_lifecycle_node->get_current_state());
+  traj_controller->on_activate(traj_lifecycle_node->get_current_state());
+
+  auto future_handle = std::async(
+    std::launch::async, [&executor]() -> void {
+      executor.spin();
+    });
+
+  typedef joint_trajectory_controller::JointTrajectoryController::ControllerStateMsg Msg;
+
+  static const int qos_level = 10;
```

Comment:
why is this `static`?

Reply:
Initialized only once, though the compiler will likely optimize it out

Reply:
initialize once. Though compiler optimizations will take care of that. It's a habit of mine though, I'll remove it.